rfd/0097-license-expiry.md

```diff
@@ -0,0 +1,147 @@
+---
+authors: camh (camh@goteleport.com)  
+state: draft
+---
+
+# RFD 97 - License Expiry
+
+## Required approvers
+
+- Engineering: `@r0mant`
+- Product:
+- Security:
+
+## What
+
+Disable SSO enterprise features after license has expired, with a grace period.
+
+## Why
+
+To encourage users to renew their Teleport Enterprise license.
+
+[RFD 84] specified a license expiry warning to encourage users to renew their
+license. This RFD extends that to disable enterprise features if the license
+still has not been renewed after a grace period.
+
+This RFD proposes to disable SAML and OIDC SSO logins only as they are likely
+the most common reason to use Teleport Enterprise. It does not seem likely that
+other enterprise features are being used without also using the enterprise SSO
+connectors, so it is unnecessary work to disable other features.
```

Comment:
I'm fine with this as a starting point, but I wouldn't say properly enforcing the license is unnecessary work. Nothing to stop someone with an expired license from creating local users and continuing to use enterprise features otherwise.

This shouldn't be too hard to do since most of these checks are centralized in `modules.GetModules()`.